js/util/process.js

```diff
@@ -0,0 +1,42 @@
+/*
+Helper class to control child process output.
+Bufferizes output from stdout and stderr, waits until the process exits,
+and then resolves the promise with gathered data.
+*/
+
+var { spawn } = require('child_process')
+
+class ProcessSpawner {
+  constructor(command, args) {
+    this.command = command
```

Comment:
We're passing potentially untrusted input (such as domain names) to this, so there needs to be some kind of filtering in order to prevent running arbitrary commands. Based on the session key, I would suggest something like `replace(/[^a-zA-Z0-9+/=.-]/, '')` on each item in the array by default, and then use a special "escape" method on the domain names that removes anything that's not letter/number/period.

Reply:
Ok, it makes sense, but

1. Node's `spawn` seems to already escape it's parameters. It's not possible to execute, for example, something like `echo 0 && cat /etc/passwd` with `spawn('echo', ['0', '&&', 'cat', '/etc/passwd'])`.
2. This class is intended to execute whatever is passed to it. If we want some additional security, I think it should be done at places where it is used. In our case: inside Bitwarden PM code.

Reply:
Oh, cool, I didn't know that; just sanitizing the domain should be fine then.